extensions/sns/build.rs

```diff
@@ -0,0 +1,34 @@
+use std::path::PathBuf;
+
+const REPLICA_REV: &str = "90e2799c255733409d0e61682685afcc2431c928";
+
+const BINARY_DEPENDENCIES: &[(&str, &str)] = &[
+    // (downloaded binary name, renamed binary name)
+    ("sns", "sns-cli"),
+];
+
+fn main() {
+    // keep copy of the dependency in the root of the project, so that cargo-dist will be able to package it into a tarball
+    let manifest_dir = PathBuf::from(env!("CARGO_MANIFEST_DIR"));
+    // and also in `target/debug` or `target/release` for development purposes (e.g. cargo run)
+    let target_dir = manifest_dir
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .join("target")
+        .join(std::env::var("PROFILE").unwrap());
```

Comment:
> PROFILE — release for release builds, debug for other builds. This is determined based on if the [profile](https://doc.rust-lang.org/cargo/reference/profiles.html) inherits from the [dev](https://doc.rust-lang.org/cargo/reference/profiles.html#dev) or [release](https://doc.rust-lang.org/cargo/reference/profiles.html#release) profile. Using this environment variable is not recommended. Using other environment variables like OPT_LEVEL provide a more correct view of the actual settings being used.

> Using this environment variable is not recommended. 

Is there a `CARGO_*` environment variable that is the whole directory we need here?

Reply:
Unfortunately, no: https://github.com/rust-lang/cargo/issues/9661
I printed every envvar available during build script with
```rust
    for (key, value) in std::env::vars() {
        println!("{}: {}", key, value);
    }
```
and there is not much to choose from :(. The only other good option is taking a parent.parent.parent of `OUT_DIR`:
```
  CARGO: /Users/mnl/.rustup/toolchains/stable-aarch64-apple-darwin/bin/cargo
  CARGO_CFG_PANIC: unwind
  CARGO_CFG_TARGET_ARCH: aarch64
  CARGO_CFG_TARGET_ENDIAN: little
  CARGO_CFG_TARGET_ENV:
  CARGO_CFG_TARGET_FAMILY: unix
  CARGO_CFG_TARGET_FEATURE: aes,crc,dit,dotprod,dpb,dpb2,fcma,fhm,flagm,fp16,frintts,jsconv,lor,lse,neon,paca,pacg,pan,pmuv3,ras,rcpc,rcpc2,rdm,sb,sha2,sha3,ssbs,vh
  CARGO_CFG_TARGET_HAS_ATOMIC: 128,16,32,64,8,ptr
  CARGO_CFG_TARGET_OS: macos
  CARGO_CFG_TARGET_POINTER_WIDTH: 64
  CARGO_CFG_TARGET_VENDOR: apple
  CARGO_CFG_UNIX:
  CARGO_ENCODED_RUSTFLAGS:
  CARGO_HOME: /Users/mnl/.cargo
  CARGO_MAKEFLAGS: -j --jobserver-fds=12,13 --jobserver-auth=12,13
  CARGO_MANIFEST_DIR: /Users/mnl/Desktop/dfx-extensions/extensions/sns
  CARGO_PKG_AUTHORS: DFINITY Team
  CARGO_PKG_DESCRIPTION:
  CARGO_PKG_HOMEPAGE:
  CARGO_PKG_LICENSE: Apache-2.0
  CARGO_PKG_LICENSE_FILE:
  CARGO_PKG_NAME: sns
  CARGO_PKG_README: README.md
  CARGO_PKG_REPOSITORY: https://github.com/dfinity/dfx-extensions
  CARGO_PKG_RUST_VERSION:
  CARGO_PKG_VERSION: 0.1.0
  CARGO_PKG_VERSION_MAJOR: 0
  CARGO_PKG_VERSION_MINOR: 1
  CARGO_PKG_VERSION_PATCH: 0
  CARGO_PKG_VERSION_PRE:
  DEBUG: true
  DYLD_FALLBACK_LIBRARY_PATH: /Users/mnl/Desktop/dfx-extensions/target/debug/deps:/Users/mnl/Desktop/dfx-extensions/target/debug:/Users/mnl/.rustup/toolchains/stable-aarch64-apple-darwin/lib/rustlib/aarch64-apple-darwin/lib:/Users/mnl/.rustup/toolchains/stable-aarch64-apple-darwin/lib:/Users/mnl/lib:/usr/local/lib:/usr/lib:/Users/mnl/Desktop/dfx-extensions/target/debug/build/libloading-482d06ac5b900652/out:/Users/mnl/Desktop/dfx-extensions/target/debug/build/libz-ng-sys-0f0fa38f8a3c41b2/out/lib:/Users/mnl/Desktop/dfx-extensions/target/debug/build/libz-ng-sys-0f0fa38f8a3c41b2/out/lib64:/Users/mnl/Desktop/dfx-extensions/target/debug/build/psm-9ab121517395d35c/out:/Users/mnl/Desktop/dfx-extensions/target/debug/build/ring-483e819f595b7e9c/out
  HOME: /Users/mnl
  HOST: aarch64-apple-darwin
  OLDPWD: /Users/mnl
  OPT_LEVEL: 0
  OUT_DIR: /Users/mnl/Desktop/dfx-extensions/target/debug/build/sns-5365b70adbf9ff15/out
  PROFILE: debug
  PWD: /Users/mnl/Desktop/dfx-extensions
  RUSTC: /Users/mnl/.rustup/toolchains/stable-aarch64-apple-darwin/bin/rustc
  RUSTDOC: /Users/mnl/.rustup/toolchains/stable-aarch64-apple-darwin/bin/rustdoc
  RUSTUP_HOME: /Users/mnl/.rustup
  RUSTUP_TOOLCHAIN: stable-aarch64-apple-darwin
  RUST_RECURSION_COUNT: 1
  TARGET: aarch64-apple-darwin
```
(I removed system-related envvars)

Reply:
I went with `OUT_DIR`, since `PROFILE` is not recommended (which I didn't know, thanks)